wdl-cli/src/inputs.rs

```diff
@@ -0,0 +1,499 @@
+    Pair {
+        /// The key.
+        key: String,
+
+        /// The value.
+        value: Value,
+    },
+}
+
+impl Input {
+    /// Attempts to return a reference to the inner [`Path`].
+    ///
+    /// * If the input is a [`Input::File`], a reference to the inner path is
+    ///   returned wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn as_file(&self) -> Option<&Path> {
+        match self {
+            Input::File(p) => Some(p.as_path()),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and attempts to return the inner [`PathBuf`].
+    ///
+    /// * If the input is a [`Input::File`], the inner path buffer is returned
+    ///   wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn into_file(self) -> Option<PathBuf> {
+        match self {
+            Input::File(p) => Some(p),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and returns the inner [`PathBuf`].
+    ///
+    /// # Panics
+    ///
+    /// If the input is not a [`Input::File`].
+    pub fn unwrap_file(self) -> PathBuf {
+        match self {
+            Input::File(p) => p,
+            v => panic!("{v:?} is not an `Input::File`"),
+        }
+    }
+
+    /// Attempts to return a reference to the inner key-value pair.
+    ///
+    /// * If the input is a [`Input::Pair`], a reference to the inner key and
+    ///   value is returned wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn as_pair(&self) -> Option<(&str, &Value)> {
+        match self {
+            Input::Pair { key, value } => Some((key.as_str(), value)),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and attempts to return the inner key-value pair.
+    ///
+    /// * If the input is a [`Input::Pair`], the inner key-value pair is
+    ///   returned wrapped in [`Some`].
+    /// * Otherwise, [`None`] is returned.
+    pub fn into_pair(self) -> Option<(String, Value)> {
+        match self {
+            Input::Pair { key, value } => Some((key, value)),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and returns the inner key-value pair.
+    ///
+    /// # Panics
+    ///
+    /// If the input is not a [`Input::Pair`].
+    pub fn unwrap_pair(self) -> (String, Value) {
+        match self {
+            Input::Pair { key, value } => (key, value),
+            v => panic!("{v:?} is not an `Input::Pair`"),
+        }
+    }
+}
+
+impl FromStr for Input {
+    type Err = Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Error> {
+        if !s.contains("=") {
+            let path = PathBuf::from(s);
+
+            if !path.exists() {
+                return Err(Error::FileNotFound(path));
+            }
+
+            return Ok(Input::File(path));
+        }
+
+        // SAFETY: we just checked above to make sure the string contains an
+        // equals sign.
+        let (key, value) = s.split_once("=").unwrap();
```

Comment:
Nit: could we just do the following to avoid the `contains` followed by `unwrap`:

```rust
match s.split_once("=") {
     Some((key, value)) => {
        // ...
     }
     None => {
        // Do the `Input::File` branch above
}
```